int32 ID
int32 centre_x
int32 centre_y
